core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -26,74 +34,97 @@
 /**
  * Container which launches Docker Compose, for the purposes of launching a defined set of containers.
  */
-public class DockerComposeContainer<SELF extends DockerComposeContainer<SELF>> extends GenericContainer<SELF> implements LinkableContainer {
+public class DockerComposeContainer<SELF extends DockerComposeContainer<SELF>> extends FailureDetectingExternalResource {
 
     /**
      * Random identifier which will become part of spawned containers names, so we can shut them down
      */
     private final String identifier;
     private final Map<String, AmbassadorContainer> ambassadorContainers = new HashMap<>();
+    private final File composeFile;
     private Set<String> spawnedContainerIds;
+    private Map<String, Integer> scalingPreferences = new HashMap<>();
+    private DockerClient dockerClient;
+
+    /**
+     * Properties that should be passed through to all Compose and ambassador containers (not
+     * necessarily to containers that are spawned by Compose itself)
+     */
+    private Map<String, String> env = new HashMap<>();
+
+    private static final RateLimiter AMBASSADOR_CREATION_RATE_LIMITER = RateLimiterBuilder
```

Comment:
Handle ambassador container startup retry here rather than in GenericContainer, now.